src/utils/__mocks__/delay.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * @desc Promise that will resolve after the ms interval
+ */
+export const delay = jest.fn().mockImplementation(
+  (ms: number): Promise<void> => {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+);
```

Comment:
Basically same function, but wrapped in a mock so we can spy on when it's called